app/controllers/studentLoans/StudentLoansCYAController.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.studentLoans
+
+import config.{AppConfig, ErrorHandler}
+import controllers.predicates.{AuthorisedAction, InYearAction, TaxYearAction}
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.studentLoans.StudentLoansCYAService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import utils.Clock
+import views.html.studentLoans.StudentLoansCYAView
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class StudentLoansCYAController @Inject()(
+                                           mcc: MessagesControllerComponents,
+                                           view: StudentLoansCYAView,
+                                           service: StudentLoansCYAService,
+                                           authAction: AuthorisedAction,
+                                           inYearAction: InYearAction,
+                                           errorHandler: ErrorHandler,
+                                           implicit val appConfig: AppConfig,
+                                           implicit val ec: ExecutionContext,
+                                           implicit val clock: Clock
+                                         ) extends FrontendController(mcc) with I18nSupport {
+
+  def show(taxYear: Int, employmentId: String): Action[AnyContent] = (authAction andThen TaxYearAction.taxYearAction(taxYear)).async { implicit request =>
+
+    if(appConfig.studentLoansEnabled) {
+      val inYear: Boolean = inYearAction.inYear(taxYear)
+      
+      service.retrieveCyaDataAndIsCustomerHeld(taxYear, employmentId)(_.fold(
+        Redirect(appConfig.incomeTaxSubmissionOverviewUrl(taxYear))
+      ) { case (cya, isCustomer) =>
+        Ok(view(taxYear, employmentId, cya, !isCustomer, inYear))
```

Comment:
Is the boolean case here reflective of the final logic? I feels a weird to use "!isCustomer" on a case with customerData? If i've misunderstood what this is doing please ignore.